src/api/envelopes_api.rs

```diff
@@ -0,0 +1,45 @@
+use super::{
+    errors::{ApiErrorKind, ApiResult},
+    Api, ApiResponse, Method,
+};
+use crate::constants::USER_AGENT;
+use log::debug;
+use sentry::{protocol::EnvelopeItem, types::Dsn, Envelope};
+use std::sync::Arc;
+
+pub struct EnvelopesApi {
+    api: Arc<Api>,
+    dsn: Dsn,
+}
+
+impl EnvelopesApi {
+    pub fn try_new() -> ApiResult<EnvelopesApi> {
+        let api = Api::current();
+        api.config
+            .get_dsn()
+            .map(|dsn| EnvelopesApi { api, dsn })
+            .map_err(|_| ApiErrorKind::DsnMissing.into())
+    }
+
+    pub fn send_item(&self, item: EnvelopeItem) -> ApiResult<ApiResponse> {
```

Comment:
nit: Just a personal preference, I would rather prefer to have a method to build the `Envelope` given a set of `EnvelopeItem`(s) and then send it directly.